### DISCUSSION TITLE
How can I center graphviz-generated figures for beamer output?

### DISCUSSION THREAD

**Discussion by felixvuo** [quarto-dev/quarto-cli#8948] [Q&A] [answered]
### Description

I'd like my graphviz diagrams to be centered but can't find an option and can't quite figure out how to do a filter.

When I used the `--debug` option and look at the generated tex file I see that other figures which also use `\includegraphics` have a `\centering` command prepended, but this does not happen for graphviz stuff or inside columns.

Am I missing an option, or is a filter the only way to achieve this?  Not feeling too confident about imeplementing one.

**Comment by cscheid** [comment]
Can you give a small complete .qmd example where you're seeing the problem?

**Reply by felixvuo** [reply]
Sure, thanks for taking a look.
Unfortunately github doesn't like me to upload .qmd so I have renamed it to .md


[mwe-center.md](https://github.com/quarto-dev/quarto-cli/files/14463506/mwe-center.md)
[mwe-center.pdf](https://github.com/quarto-dev/quarto-cli/files/14463513/mwe-center.pdf)

**Reply by cscheid** [reply, marked as answer]
I'm not a beamer expert, but you can inspect the `keep-md: true` output (it'll be named `mwe-center.beamer.md`), and in a simplified version of your file, I see this (running quarto from our `main` branch)

````
# Introduction

## Expected Utility Revisited - Expected Outcome




::::::{.cell fig-width="1.8" layout-align="default"}

:::::{.cell-output-display}

::::{}
`<figure class=''>`{=html}

:::{}

![](mwe-center_files/figure-beamer/dot-figure-1.png){width="1.8in" height="7in" fig-pos='H' fig-env='figure'}
:::
`</figure>`{=html}
::::
:::::
::::::




## Expected Utility Revisited - Expected Outcome

![](https://placeholder.co/400)
````

You can see that Quarto simply produces an image for the dot diagram, and that an image is not centered in Beamer by default (I added the placeholder.co image for comparison, and that isn't centered either).

The easiest way to center it is to make the image a cross-referenceable figure:

````
```{dot}
//| fig-width: 1.8
//| fig-align: center
//| label: fig-1
//| fig-cap: a diagram.
...
```
````

(note that you had a space between `//` and `|`, and that space shouldn't be there.) Unfortunately, in LaTeX that means you'll get a `Figure 1: a diagram` text underneath the diagram, which is not really avoidable.

The brute-force way to center an image without turning it into a figure is to surround your image with `` `{\centering`{=latex} `` and `` `}`{=latex}` `` commands.

We should probably make that easier on beamer users.

**Reply by felixvuo** [reply]
That's great, thank you.  The addition of `label: fig-blah` seems to make the difference.  Yes, it forces a caption, but that's ok, figures should be captioned :-)

Thanks so much for the quick reply - I am newbie to quarto, and it's proving really helpful for producing lecture slides in multiple outputs and git-friendly too.

**Reply by felixvuo** [reply]
One great thing would be if fig-alt worked too for graphviz-generated images.  My main reason for using quarto is to be able to produce more accessible formats for lecture slides (in this case docx because html uploads are forbidden by the vle).  Unfortunately the images produced in the docx file are both stretched and without alt text.  Is the intention that a `//| fig-alt: "my description"` should result in an alternative text data in the resulting document?

**Reply by mcanouil** [reply]
@felixvuo I suggest you open a new discussion (after checking none is already covering this) as this last request is unrelated to the current thread.
